### PR TITLE
Alert manager design doc

### PR DESCRIPTION
**Pull request by sushanthakumar**
**What type of PR is this?**
 /kind design

**What this PR does / why we need it**:
This PR adds Alert manager design doc

**Which issue(s) this PR fixes**:

Fixes #

**Test Report Added?**:
 /kind NOT-TESTED

**Test Report**:

**Special notes for your reviewer**:

### CODE REVIEW

File: specs/SIM/alert_manager/SODA_AlertManagerDesign.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+|                |              |              | 6:Alarm unacknowledgement                |                  |                                          |
+|                |              |              | 7:Other change                           |                  |                                          |
+| occurUtcTime   | long         | milliseconds |                                          | Mandatory        | UTC time when the alarm is generated (in milliseconds) |
+| matchKey       | String(255)  |              |                                          | Mandatory        | Alarm matching key                       |
+| meDn           | String(255)  |              |                                          | Not              | Unique Id of alarm source device in the whole management system. |
+| meName         | String(255)  |              |                                          | Mandatory        | Alarm source which identifies the device that generates an alarm. Alarm source can be identified by the device name and IP address. |
```
Comment (NajmudheenCT) [inline specs/SIM/alert_manager/SODA_AlertManagerDesign.md:195]: is this name final "meName" "MeDn" etc..

File: specs/SIM/alert_manager/SODA_AlertManagerDesign.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+|                |              |              | 6: System Clearance(system clearance due to management capacity limitations) |                  |                                          |
+|                |              |              | 7: Synchronous clearance(Clearance due to interconnected system) |                  |                                          |
+| meCategory     | Enum(String) |              |                                          | Mandatory        | Resource category                        |
+|                |              |              | Network                                  |                  |                                          |
+|                |              |              | Server                                   |                  |                                          |
+|                |              |              | Storage                                  |                  |                                          |
+
+
+### Sequence Diagrams
+* Configuring alert source
```
Comment (kumarashit) [inline specs/SIM/alert_manager/SODA_AlertManagerDesign.md:226]: Configuring Alert source doesn't have any interaction with Device Manager?

File: specs/SIM/alert_manager/SODA_AlertManagerDesign.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+| Remarks     | None|
+
+
+#### Non Functional Requirements
+NA
+
+## Architecture Analysis
+### SODA Infrastructure Manager: High Level View
+![](../resources/Architecture.jpg)
+
```
Comment (kumarashit) [inline specs/SIM/alert_manager/SODA_AlertManagerDesign.md:123]: What is our considerations?
- Can the Alert Manager and Driver Manager be separate services running independently? Say alert Manager is running on node1 and device manager on Node2..
- The duty of "Alert config" is just to trap config and clear alert or something more? 
- Can alert config be part of Device Manager as the task currently it is doing is just to be a router

File: specs/SIM/alert_manager/SODA_AlertManagerDesign.md (hunk 1 above)
Comment (sushanthakumar) [reply specs/SIM/alert_manager/SODA_AlertManagerDesign.md]: Udpated in latest commit

File: specs/SIM/alert_manager/SODA_AlertManagerDesign.md (hunk 2 above)
Comment (sushanthakumar) [reply specs/SIM/alert_manager/SODA_AlertManagerDesign.md:256]: No configuration at device side

**Review by NajmudheenCT — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Pull request merged**